compiler/qsc_rca/tests/binops.rs

```diff
@@ -0,0 +1,133 @@
+    );
+    let package_store_compute_properties = compilation_context.get_compute_properties();
+    check_last_statement_compute_properties(
+        package_store_compute_properties,
+        &expect![
+            r#"
+            ApplicationsGeneratorSet:
+                inherent: Quantum: QuantumProperties:
+                    runtime_features: RuntimeFeatureFlags(UseOfDynamicBool)
+                    value_kind: Element(Dynamic)
+                dynamic_param_applications: <empty>"#
+        ],
+    );
+}
+
+#[test]
+fn check_rca_for_bin_op_with_classical_lhs_and_dynamic_rhs() {
+    let mut compilation_context = CompilationContext::new();
+    compilation_context.update(
+        r#"
+        use q = Qubit();
+        One == M(q)"#,
+    );
+    let package_store_compute_properties = compilation_context.get_compute_properties();
+    check_last_statement_compute_properties(
+        package_store_compute_properties,
+        &expect![
+            r#"
+            ApplicationsGeneratorSet:
+                inherent: Quantum: QuantumProperties:
+                    runtime_features: RuntimeFeatureFlags(UseOfDynamicBool)
+                    value_kind: Element(Dynamic)
+                dynamic_param_applications: <empty>"#
+        ],
+    );
+}
+
+#[test]
+fn check_rca_for_bin_op_with_dynamic_lhs_and_dynamic_rhs() {
+    let mut compilation_context = CompilationContext::new();
+    compilation_context.update(
+        r#"
+        use (a, b) = (Qubit(), Qubit());
+        let (c, d) = (M(a) == Zero, M(b) == Zero);
+        c and d"#,
+    );
+    let package_store_compute_properties = compilation_context.get_compute_properties();
+    check_last_statement_compute_properties(
+        package_store_compute_properties,
+        &expect![
+            r#"
+            ApplicationsGeneratorSet:
+                inherent: Quantum: QuantumProperties:
+                    runtime_features: RuntimeFeatureFlags(UseOfDynamicBool)
+                    value_kind: Element(Dynamic)
+                dynamic_param_applications: <empty>"#
+        ],
+    );
+}
+
+#[test]
+fn check_rca_for_nested_bin_ops_with_classic_operands() {
+    let mut compilation_context = CompilationContext::new();
+    compilation_context.update(
+        r#"
+        open Microsoft.Quantum.Math;
+        Sin(PI() / 2.0) ^ 2.0 + Cos(PI() / 2.0) ^ 2.0"#,
+    );
+    let package_store_compute_properties = compilation_context.get_compute_properties();
+    check_last_statement_compute_properties(
+        package_store_compute_properties,
+        &expect![
+            r#"
+            ApplicationsGeneratorSet:
+                inherent: Classical
+                dynamic_param_applications: <empty>"#
+        ],
+    );
+}
+
+#[test]
+fn check_rca_for_nested_bin_ops_with_a_dynamic_operand() {
+    let mut compilation_context = CompilationContext::new();
+    compilation_context.update(
+        r#"
+        use q = Qubit();
+        let i = M(q) == Zero ? 0 | 1;
```

Comment:
food for thought: if you want to reduce the complexity of some tests that require a dynamic integer you could use `DrawRandomInt` instead of relying on branching from qubit measurement. That said, the qubit measurement is more indicative of the kind of dynamism programs will use, so it's a trade off.

Reply:
For tests I would prefer to use explicit qubit measurements as sources of dynamism to make it clear for people that might not be as familiar with the implementation details of RCA.